ArcGISRuntimeSDKQt_CppSamples/Maps/ApplyScheduledMapUpdates/ApplyScheduledMapUpdates.cpp

```diff
@@ -0,0 +1,216 @@
+    const QString srcItemPath = srcPath + "/" + info.fileName();
+    const QString dstItemPath = dstPath + "/" + info.fileName();
+    if (info.isDir()) {
+      if (!copyDir(srcItemPath, dstItemPath)) {
+        return false;
+      }
+    } else if (info.isFile()) {
+      if (!QFile::copy(srcItemPath, dstItemPath)) {
+        return false;
+      }
+    } else {
+      qDebug() << "Unhandled item" << info.filePath() << "in copyDir";
+    }
+  }
+  return true;
+}
+}
+
+ApplyScheduledMapUpdates::ApplyScheduledMapUpdates(QObject* parent /* = nullptr */):
+  QObject(parent)
+{
+  // For the purposes of demonstrating the sample,
+  // create a temporary copy of the local offline map files,
+  // so that updating does not overwrite them permanently
+  copyDir(defaultDataPath() + "/" + sampleMmpk, m_TempDir.path() + "/" + sampleMmpk);
+}
+
+ApplyScheduledMapUpdates::~ApplyScheduledMapUpdates()
+{
+  if (m_mobileMapPackage)
+  {
+    m_mobileMapPackage->close();
+  }
+}
+
+void ApplyScheduledMapUpdates::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<ApplyScheduledMapUpdates>("Esri.Samples", 1, 0, "ApplyScheduledMapUpdatesSample");
+}
+
+void ApplyScheduledMapUpdates::updateMap()
+{
+  if (!m_offlineSyncTask)
+    return;
+
+  m_offlineSyncTask->createDefaultOfflineMapSyncParameters();
+}
+
+MapQuickView* ApplyScheduledMapUpdates::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void ApplyScheduledMapUpdates::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+
+  // create MMPK
+  qDebug() << m_TempDir.path() + "/" + sampleMmpk;
+  m_mobileMapPackage = new MobileMapPackage(m_TempDir.path() + "/" + sampleMmpk, this);
+
+  // load MMPK
+  connect(m_mobileMapPackage, &MobileMapPackage::doneLoading, this, [this](Error e)
+  {
+    if (!e.isEmpty())
+      return;
+
+    // set the map on the map view
+    m_map = m_mobileMapPackage->maps().at(0);
+    m_mapView->setMap(m_map);
+
+    // setup sync task
+    m_offlineSyncTask = new OfflineMapSyncTask(m_map, this);
+
+    // connect sync task signals
+    connectSyncSignals();
+
+    // check for updates
+    m_offlineSyncTask->checkForUpdates();
+  });
+
+  m_mobileMapPackage->load();
+
+  emit mapViewChanged();
+}
+
+void ApplyScheduledMapUpdates::connectSyncSignals()
+{
+  // connect to checkForUpdatesCompleted signal and update the UI accordingly
+  connect(m_offlineSyncTask, &OfflineMapSyncTask::checkForUpdatesCompleted, [this](QUuid, OfflineMapUpdatesInfo* info)
+  {
+    if (info->downloadAvailability() == OfflineUpdateAvailability::Available)
+    {
+      emit updateUi(true, "Updates Available", QString("Updates size: %1 bytes").arg(info->scheduledUpdatesDownloadSize()));
+    }
+    else
+    {
+      emit updateUi(false, "No updates available", "The preplanned map area is up to date");
+    }
+
+    delete info;
```

Comment:
good thinking to clean this up. Do we ever use any kind of smart pointer or RAII in samples to ensure these get cleaned up? May also be worth a comment to explain?

Reply:
we've mainly used QScopedPointer - https://github.com/Esri/arcgis-runtime-samples-qt/blob/37f29d3634a2ee71933836e6546c58d42fcfe081/ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/IdentifyGraphics/IdentifyGraphics.cpp#L111. We've used a few other techniques too I believe, depending on the scenario. 

This one was very straight forward because I only use the info to make the one check, and then I'm done, so it's safe to get rid of right away. Most other scenarios, we might hold onto the pointer for a bit to do some things. I can change this to QScopedPointer (or something else) if you think it is better